internal/lifecycle/lifecycle.go

```diff
@@ -77,6 +77,11 @@ func (l *Lifecycle) Start(ctx context.Context) error {
 	l.mu.Unlock()
 
 	for _, hook := range l.hooks {
+		// if ctx has cancelled, bail out of the loop.
+		if err := ctx.Err(); err == context.Canceled || err == context.DeadlineExceeded {
+			return err
+		}
```

Comment:
```suggestion
		if err := ctx.Err(); errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
			return err
		}
```

Should we use errors.Is?